Talk-of-the-network.ipynb

```diff
@@ -0,0 +1,649 @@
+    "\n",
+    "- What matters more - strong ties or weak ties?\n",
+    "- What effect does the size of an average individuals network have?\n",
+    "- How does advertising interact with the diffusion through weak ties and that through strong ties\n",
+    "\n",
+    "In this workbook, we focus on replicating the efforts of the authors to answer the first question: do strong ties or weak ties influence the speed of information dissemination in a network?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2. Initializing the network"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This study employs a large number of synthetic networks as substrates to study the diffusion of information diffusion. To quote the authors logic to create and initialize the networks:\n",
+    "\n",
+    "> *\"Each individual belongs to a single personal network. Each network consists of individuals who are connected by strong ties. In each period, individuals also conduct a finte number of weak tie interactions outside their personal networks... We divide the entire market equally into personal networks, in which each individual can belong to one network. In addition, in each period, every individual conducts random meetings with individuals external to his personal network.\"*\n",
+    "\n",
+    "Given this specification, we utilize the built-in complete graph generator from [LightGraphs](http://juliagraphs.github.io/LightGraphs.jl/latest/generators.html#Graph-Generators-1) to build several mini-regular networks and then allow individuals in each of these mini-networks to mingle. Our final data structure is hence a dictionary of several complete networks that are built based on the number of strong ties for each individual. Note that each individual in the network has a fixed number of strong ties ($s$) and weak ties ($w$)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "initialize_network (generic function with 1 method)"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "function initialize_network(n_nodes::Int, n_strong_ties::Int)\n",
+    "    G = Dict{Int, LightGraphs.SimpleGraphs.SimpleGraph}()\n",
+    "\n",
+    "    for g in 1:floor(Int, n_nodes/n_strong_ties)\n",
+    "        G[g] = CompleteGraph(n_strong_ties)\n",
+    "    end\n",
```

Comment:
Why use a dictionary here? Why not a vector?

Reply:
Sure, I started with a vector, but realized that for all the sub-networks the vertex ids from `vertices(g)` has the same range , i.e., `1:number_of_nodes`.  A dictionary seemed to be clearer to read/understand since each sub-network has its own id and the associated nodes, even though have the same ids are cleanly separated. Once the overall network was a dictionary, it made sense that the `node_status` was a dictionary too. Overall, I felt being explicit about the sub-network we are referring to makes the code further down the line, i.e., random sampling, weak tie mixing, etc clearer.